pandera/backends/polars/checks.py

```diff
@@ -46,7 +46,25 @@ def preprocess(self, check_obj: pl.LazyFrame, key: Optional[str]):
 
     def apply(self, check_obj: PolarsData):
         """Apply the check function to a check object."""
-        return self.check_fn(check_obj)
+        out = self.check_fn(check_obj)
+
+        if isinstance(out, bool):
+            return out
+
+        if len(out.columns) > 1:
```

Comment:
I think after my changes in https://github.com/unionai-oss/pandera/pull/1518, all checks return a single column. So do we still need this condition, or is this still necessary for custom checks?

Reply:
yeah this is to support custom dataframe-level checks that may produce a boolean dataframe